src/main/java/org/truffleruby/interop/InteropNodes.java

```diff
@@ -1000,4 +1005,48 @@ public int keyInfoFlagsToBitsNode(boolean readable, boolean invocable, boolean i
 
     }
 
+    @CoreMethod(names = "java_type", isModuleFunction = true, required = 1)
+    public abstract static class JavaTypeNode extends CoreMethodArrayArgumentsNode {
+
+        @TruffleBoundary
+        @Specialization(guards = "isRubySymbol(name)")
+        public Object javaTypeSymbol(DynamicObject name) {
+            return javaType(name.toString());
+        }
+
+        @TruffleBoundary
+        @Specialization(guards = "isRubyString(name)")
+        public Object javaTypeString(DynamicObject name) {
+            return javaType(name.toString());
```

Comment:
It's clearer to use Layouts.SYMBOL.getString() here.

Reply:
Fixed.